Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -459,6 +459,11 @@ void doEsirkepovDepositionShapeN (const GetParticlePosition& GetPosition,
             ParticleReal xp, yp, zp;
             GetPosition(ip, xp, yp, zp);
 
+            // Modify the particle position to match the time of the deposition
+            xp += relative_time * uxp[ip] * gaminv;
+            yp += relative_time * uyp[ip] * gaminv;
+            zp += relative_time * uzp[ip] * gaminv;
```

Comment:
Doing these preliminary shifts here (instead of combining `relative_time` with `0.5_rt*dt` later when the new and old coordinates are computed) seems to make some single-precision checksum benchmarks fail, in the case of Esirkepov deposition (Vay deposition is probably not tested in single precision).

Reply:
I think https://github.com/ECP-WarpX/WarpX/pull/2937/commits/8c49d5a7ea738b324729278673073f9c7c124cd3 fixes the following checksums:
- `ionization_boost`
- `RepellingParticles`
- `LaserAccelerationBoost`
- `PEC_particle`
- `LaserAcceleration_single_precision_comms`
- `reduced_diags_single_precision`